pandas/core/series.py

```diff
@@ -207,10 +211,20 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                 else:
                     data = data.reindex(index, copy=copy)
                 data = data._data
-            elif isinstance(data, dict):
+            elif isinstance(data, dict) and (len(data) or index is None):
```

Comment:
rather than doing this here, why not inside ``_init_dict`` itself? (you can simply return early if these conditions ar met). you can also return the ``type, copy`` parameter as well. ideally like to locate all of a single 'type' code in the same place (dict here)

Reply:
I agree - in particular, ``_init_dict`` already checks whether ``data`` is empty